### PR TITLE
updated README.md, added CONTRIBUTING.md, updated keip-container-arch…

### PR DESCRIPTION
**Pull request by dcruver**
…etype/README.md

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,65 +1,46 @@
...
-   in cases where
-   custom components are required.
-2. operator -
-    1. resource - this is the Kubernetes custom resource definition, `IntegrationRoute`.
-    2. operator - this is the operator which does the cluster work in reaction to the creation
-       of `IntegrationRoutes`.
+## Key Features
+- **Kubernetes Native**: Fully integrates with Kubernetes, using custom resources to manage Spring Integration routes.
+- **Ease of Use**: Allows the definition and deployment of complex integration patterns with simple Kubernetes manifests.
+- **Flexibility**: Supports a range of configurations to cater to different integration requirements.
```
Comment (a-asaad) [inline README.md:13]: :question: Should we highlight the runtime configuration capabilities here?

File: README.md [hunk 2]
```diff
@@ -69,15 +50,75 @@ NAME                                        READY   STATUS    RESTARTS   AGE
...
+Note: If this is your first time through this installation process, it may be helpful to return to this step later. This
+is generally needed for all but the simplest deployments, but it isn't necessary if you're just familiarizing yourself 
+with the installation process.
+
+The default keip container provides only the basic components for Spring Integration. To fully utilize the potential of 
+your integration routes, you will need to include additional Spring Integration components or your own Java code. See 
+[README.md](keip-container-archetype%2FREADME.md) for instructions on how to create a custom container.
+
+Once your new container is available, you'll need to set that name in the keip-controller-props ConfigMap. The name of 
+this ConfigMap includes a hash, so we'll need to retrieve the full ConfigMap name first. We'll need to change the value
```
Comment (a-asaad) [inline README.md:63]: The `make controller/deploy` command seems to generate the configmap using a plain `kubectl create cm keip-controller-props`, so I'm not sure a hash suffix is actually generated.

File: README.md [hunk 3]
```diff
@@ -69,15 +50,75 @@ NAME                                        READY   STATUS    RESTARTS   AGE
...
+for the integration-image key.
+
+Note that the single-line command below requires that you have the `jq` utility installed.
+
+```shell
+kubectl edit cm `kubectl get configmap --all-namespaces -o json | jq -r '.items[] | select(.metadata.name | startswith("keip-controller-props")) | .metadata.name'`
```
Comment (a-asaad) [inline README.md:69]: :question: If the webhook is already running, will editing the `keip-controller-props` configmap immediately take effect? Or do we need to restart the webhook pod?

File: README.md (hunk 2 above)
Comment (dcruver) [reply README.md:63]: On my local machine, this CM is called keip-controller-props-2tmmf9gh84

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# Contributing to keip
+First off, thanks for taking the time to contribute! 🎉
+
+The following is a set of guidelines for contributing to KeIP, hosted in the OctoConsulting Organization on GitHub.
```
[the remark this answers is not in the document]
Comment (dcruver) [reply CONTRIBUTING.md:4]: fixed

File: README.md (hunk 3 above)
Comment (dcruver) [reply README.md]: Added a step to restart the webhook.

File: README.md (hunk 2 above)
Comment (dcruver) [reply README.md]: fixed

**Review by a-asaad — APPROVED** [review]

**Pull request merged**